compiler/z/codegen/ControlFlowEvaluator.cpp

```diff
@@ -3637,22 +3640,65 @@ OMR::Z::TreeEvaluator::ternaryEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-      }
+         if (comp->getOption(TR_TraceCG))
+            traceMsg(comp, "Emitting a Load on condition with condtion based on compare type\n");
+         if (trueReg->getRegisterPair() || falseReg->getRegisterPair())
+            {
+            TR_ASSERT(trueReg->getRegisterPair() && falseReg->getRegisterPair(), "Both (or none) false and true regs should be register pairs");
+            
+            const bool is64BitRegister = trueReg->getKind() == TR_GPR64 || cg->use64BitRegsOn32Bit();
+            
+            generateRRFInstruction(cg, is64BitRegister ? TR::InstOpCode::LOCGR: TR::InstOpCode::LOCR, node, trueReg->getHighOrder(), falseReg->getHighOrder(), getMaskForBranchCondition(TR::InstOpCode::COND_BERC)>>4, true);
+            generateRRFInstruction(cg, is64BitRegister ? TR::InstOpCode::LOCGR: TR::InstOpCode::LOCR, node, trueReg->getLowOrder(), falseReg->getLowOrder(), getMaskForBranchCondition(TR::InstOpCode::COND_BERC)>>4, true);
+            }
+         else
+            {
+            generateRRFInstruction(cg, trueVal->getOpCode().is8Byte() ? TR::InstOpCode::LOCGR: TR::InstOpCode::LOCR, node, trueReg, falseReg, getMaskForBranchCondition(TR::TreeEvaluator::mapBranchConditionToLOCRCondition(bc))>>4, true);
+            }
+         }
       else
-         generateRRFInstruction(cg, trueVal->getOpCode().is8Byte() ? TR::InstOpCode::LOCGR: TR::InstOpCode::LOCR, node, trueReg, falseReg, getMaskForBranchCondition(TR::TreeEvaluator::mapBranchConditionToLOCRCondition(bc))>>4, true);
+         {
+         TR::LabelSymbol *branchDestination = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+         branchDestination->setEndInternalControlFlow();
+
+         TR::Instruction *compareInst = generateS390CompareAndBranchInstruction(cg, compareOp, node, firstReg, secondReg, bc, branchDestination, false);
+
+         compareInst->setStartInternalControlFlow();
+
+         TR::RegisterDependencyConditions * conditions = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 4, cg);
+         conditions->addPostCondition(firstReg, TR::RealRegister::AssignAny);
+
+         if (secondReg != firstReg)
+            {
+            conditions->addPostCondition(secondReg, TR::RealRegister::AssignAny);
+            }
+
+         if (falseReg != firstReg && falseReg != secondReg)
+            {
+            conditions->addPostCondition(falseReg, TR::RealRegister::AssignAny);
+            }
+
+         if (trueReg != falseReg && trueReg != firstReg && trueReg != secondReg)
+            {
+            conditions->addPostCondition(trueReg, TR::RealRegister::AssignAny);
+            }
+
+         TR::Instruction *conditionalLoad = generateRRInstruction(cg, trueVal->getOpCode().is8Byte() ? TR::InstOpCode::LGR : TR::InstOpCode::LR, node, trueReg, falseReg);
```

Comment:
The z196 path has code to handle if trueReg/falseReg is a register pair.  However, the non-z196 path doesn't seem to handle this case.  Is that okay?